sql/core/src/main/scala/org/apache/spark/sql/execution/ColumnarEvaluatorFactory.scala

```diff
@@ -0,0 +1,116 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.execution
+
+import scala.collection.JavaConverters._
+
+import org.apache.spark.{PartitionEvaluator, PartitionEvaluatorFactory, TaskContext}
+import org.apache.spark.sql.catalyst.InternalRow
+import org.apache.spark.sql.catalyst.expressions.{Attribute, UnsafeProjection}
+import org.apache.spark.sql.execution.metric.SQLMetric
+import org.apache.spark.sql.execution.vectorized.{OffHeapColumnVector, OnHeapColumnVector, WritableColumnVector}
+import org.apache.spark.sql.types.StructType
+import org.apache.spark.sql.vectorized.ColumnarBatch
+
+// This avoids calling `childOutput`in the RDD closure,
+// so that we don't need to include the entire plan (this) in the closure.
```

Comment:
Because PartitionEvaluatorFactory is designed for that purpose. And the comment is for original trick. Not sure if this is still needed. Looks a bit out-of-dated now.

Reply:
If it's not out-of-dated, we should move the comment into `override def eval`.

Reply:
The original comment is for the trick to avoid issue for closure. `PartitionEvaluatorFactory` doesn't use closure so that's why I think it is out-of-dated. Actually it will be a bit confused for code readers.

Reply:
Removed outdated comment

Reply:
@viirya Thank you for the explanation.